flex/validation/common.py

```diff
@@ -349,18 +383,18 @@ def validate_object(obj, field_validators=None, non_field_validators=None,
         construct_schema_validators,
     )
     schema_validators = construct_schema_validators(schema, context)
-
     if '$ref' in schema_validators and hasattr(schema_validators['$ref'], 'validators'):
         ref_ = field_validators.pop('$ref')
         for k, v in ref_.validators.items():
             if k not in schema_validators:
                 schema_validators.add_validator(k, v)
 
-    schema_validators.update(field_validators)
+    if 'discriminator' in schema:
+        add_polymorphism_requirements(obj, schema, context, schema_validators)
```

Comment:
Same comment about mutation here.  I would prefer it if this function returned an updated `schema_validators` and overwrite the existing variable here in this function scope.